### DISCUSSION TITLE
reuse mapper is ignored

### DISCUSSION THREAD

**Discussion by jeusdi** [mapstruct/mapstruct#3563] [Q&A]
I'm trying to re-use an already defined Mapper on another one.

```java
@Mapper
public interface ReferenceUniqueIdentifierMapper {
        //...
	default String toPersistenceModel(ReferenceUniqueIdentifier id) {
		return id.getId().toString();
	}
}
```

In another Mapper, I'm trying to re-use previous defined mapper:

```java
@Mapper(uses = ReferenceUniqueIdentifierMapper.class)
public interface NestedReferenceMapper {

        //...

	@Mapping(source = "referenceUniqueIdentifier", target = ".")
	String toPersistenceModel(NestedReference persistence);
```

`Nestedrefference` is like:

```java
public class NestedReference {

	private final ReferenceUniqueIdentifier referenceUniqueIdentifier;
        //...
}
```

So, I'm trying to map `NestedReference.referenceUniqueIdentifier` to String using `ReferenceUniqueIdentifier.toPersistenceModel(ReferenceUniqueIdentifier id)` .

Generated code is:

```
    @Override
    public String toPersistenceModel(NestedReference persistence) {
        if ( persistence == null ) {
            return null;
        }

        String string = new String();

        return string;
    }
```

Any ideas about how to peak `ReferenceUniqueIdentifierMapper.toPersistenceModel` from `NestedReferenceMapper`?

**Comment by filiphr** [comment]
We do not support mapping Bean to a primitive type as you are trying to do. This means that

```java
@Mapping(source = "referenceUniqueIdentifier", target = ".")
String toPersistenceModel(NestedReference persistence);
```

Does not work as you think it does.

I have seen this a lot of times, perhaps it is time for us to support something like this. Perhaps with a new annotation.

e.g.


```java
@PrimitiveMapping(source = "referenceUniqueIdentifier")
String toPersistenceModel(NestedReference persistence);
```

Maybe a better name then `@PrimitiveMapping` as the goal would be that the target is not a bean but just a target type to map to.
